### DISCUSSION TITLE
Feedback on test spanish course

### DISCUSSION THREAD

**Discussion by ghost** [kantord/LibreLingo#2902] [General]
Don't know if I should create separate issues for each of these, but here are the things I think could be improved

Browser: firefox 113.0.2 (64-bit)

# text is off center in buttons for script symbols

![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/e3d71b7a-9853-409d-aa08-73579573e795)

# button styles don't match

There are three different styles for buttons that all don't match.

![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/5f49f006-e4a5-4446-b735-865c04156bca)

# no visual cues for buttons

![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/41aed0b0-20d1-4a39-9a9b-d793ad8bd59e)

Maybe something as simple as  ![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/f666d63b-8f3d-4cd9-a84e-a981101cf367) or ![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/08fcb678-ba40-4a0d-bea1-faedccec3bec) beside the submit button would make it apparent. 

Also numbers below the images and a little text like "press one of the numbers to select an image"

# No pronunciation for words or sentences

Maybe it is already possible to provide them and they just aren't there for the Spanish course, but if they are available, then there should be a button to play the audio.

Course creators should also be given the possibility to provide the audio themselves. One could imagine a mapping of string to audio in order to reuse it as much as possible. Or even promote course creators to contribute the pronunciation to the wiktionary and librelingo to fallback to wiktionary automatically where possible. 
Wiktionary has **a lot** of pronunciations contributed by thousands of volunteers in different languages. For example [impossible](https://fr.wiktionary.org/wiki/impossible#Prononciation) in French. Lots are missing, but this would be one way to get pronunciations for librelingo **and** contribute to wiktionary at the same time.

# Promote mastodon too

![image](https://github.com/LibreLingo/LibreLingo/assets/2829538/3d9967e9-2d49-44c4-8933-af1362b8fb4d)

Right now, only twitter is provided as an option. Mastodon is the opensource alternative and can provide a mastodon button too. [Here's a lib](https://sharetomastodon.github.io/about/)

And here's an example button
<a href="https://sharetomastodon.github.io/?title=LibreLingo app&url=https://librelingo.app" target="_blank">
		<img src="https://sharetomastodon.github.io/favicon.png" style="border: 0px solid;" />
	</a>

------------

All in all, I think it's a good start and I look forward to the development.

**Comment by kantord** [comment]
Nice! Thanks! I particularly like the idea about visual cues for buttons

**Comment by Bouaziz-aitd** [comment]
@LovelsGrief those are pertinent comments. The audio is important as there are languages that do not have any audio right now. Kantord who is leading the priorities was of the opinion that some of the developments will take a long time before they are tackled.
I am working on the the Kabyle "kab" course and I was struggling since months on how to implement any TTS. I finally have two options I was about to bring in the thread related to that (refer to that discussion for more details). I hope this time it will pick attention. Pleas note that my knowledge to implement all this in Libre Lingo is limited but I did some tests on my side to get proof the options works.
